core/src/main/java/greencity/controller/HabitController.java

```diff
@@ -87,125 +88,171 @@
-            .body(habitAssignService.assignHabitForUser(habitId, user));
+            .body(habitAssignService.assignHabitForUser(id, userVO));
     }
 
     /**
-     * Method return {@link HabitAssign} by it's id.
+     * Method returns {@link HabitAssign} by it's id.
      *
-     * @param habitAssignId - id of {@link HabitAssign}
-     * @return {@link HabitAssignDto}
+     * @param id {@link HabitAssign} id.
+     * @return {@link HabitAssignDto}.
      */
     @ApiOperation(value = "Get habit assign.")
     @ApiResponses(value = {
-        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 200, message = HttpStatuses.OK, response = HabitAssignDto.class),
         @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
-        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+        @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND)
     })
-    @GetMapping("/assign/{habitAssignId}")
-    public ResponseEntity<HabitAssignDto> getHabitAssign(@PathVariable Long habitAssignId) {
+    @GetMapping("/assign/{id}")
+    public ResponseEntity<HabitAssignDto> getHabitAssign(@PathVariable Long id) {
         return ResponseEntity.status(HttpStatus.OK)
-            .body(habitAssignService.getById(habitAssignId));
+            .body(habitAssignService.getById(id));
     }
 
     /**
-     * Method to update {@link HabitAssign} for it's id.
+     * Method to return all {@link HabitAssign} by it's {@link Habit} id.
      *
-     * @param habitAssignId - id of {@link HabitAssign}.
-     * @return {@link HabitAssignDto}.
+     * @param id       {@link Habit} id.
+     * @param acquired {@link Boolean} status.
+     * @return {@link List} of {@link HabitAssignDto}.
      */
-    @ApiOperation(value = "Update habit assign acquired or suspended status.")
+    @ApiOperation(value = "Get all user assigns from certain habit.")
     @ApiResponses(value = {
-        @ApiResponse(code = 201, message = HttpStatuses.CREATED),
-        @ApiResponse(code = 303, message = HttpStatuses.SEE_OTHER),
-        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
+        @ApiResponse(code = 200, message = HttpStatuses.OK, response = List.class),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND)
     })
-    @PatchMapping("assign/{habitAssignId}")
-    public ResponseEntity<HabitAssignDto> updateAssign(
-        @PathVariable Long habitAssignId, @Valid @RequestBody HabitAssignStatDto habitAssignStatDto) {
-        return ResponseEntity.status(HttpStatus.OK).body(habitAssignService
-            .updateStatus(habitAssignId, habitAssignStatDto));
+    @GetMapping("/{id}/assign/{acquired}")
+    public ResponseEntity<List<HabitAssignDto>> getAllHabitAssignsByHabitIdAndAcquired(@PathVariable Long id,
+                                                                                       @PathVariable Boolean acquired) {
```

Comment:
Path this variable as request param, not path variable, because it looks not understandable, why url will end tih true or false

Reply:
Changes done.